src/plugins/files/server/file_service/internal_file_service.ts

```diff
@@ -80,10 +81,33 @@ export class InternalFileService {
     }
   }
 
+  private async bulkGet(ids: string[]): Promise<IFile[]> {
+    try {
+      const metadatas = await this.metadataClient.bulkGet({ ids });
+      const result = metadatas.map(({ id, metadata }) => {
+        if (metadata.Status === 'DELETED') {
+          throw new FileNotFoundError('File has been deleted');
```

Comment:
I don't think this will provide the functionality we need in cases. Cases is intentionally doing a promise settled to allow for some files to not exist but still not have the entire api call throw an error. This way we can continue with our deletion logic even when some files don't exist. Could we still return normally here but provide a way to see which ids failed and which succeeded?